qcodes/utils/installation.py

```diff
@@ -0,0 +1,40 @@
+"""This module contains helper scripts to make certain installation tasks
+easier."""
+
+import os
+import json
+from qcodes.station import SCHEMA_PATH, update_config_schema
+
+
+def register_station_schema_with_vscode():
+    """This function registeres the qcodes station schema with vscode.
+
+    Run this function to add the user generated station schema to the list of
+    associated schemas for the Red Hat YAML schema extension for vscode.
+    (https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml)
+
+    This function will effectively add an entry to `yaml.schemas` in the user
+    config file of vscode, which is located under
+    `%APPDATA/Code/User/settings.json`.
+
+    You can alternatively access this
+    setting via File->Preferences->Settings and search for `yaml.schemas`.
+
+    To enable autocompletinon of QCoDeS instrument from additional packages run
+    `qcodes.station.update_config_schema`.
+
+    For more information consult `qcodes/docs/examples/Station.ipynb`.
+    """
+    if not os.path.exists(SCHEMA_PATH):
+        update_config_schema()
```

Comment:
i have a feeling that this "side effect" should not be part of this function. The only thing i'd expect it to do is just to raise an exception if `SCHEMA_PATH` doesn't exist. I think that implicitly updating the schema here is probably user-friendly but it does upgrade the schema "silently" may not be what user want to happen. But i'm not sure, just raising a feeling.

Reply:
I have the same feeling as you have here. But let me try to explain my thoughts and my attempt of being pragmatic:
For me it would be most preferable if that function would live somewhere outside of qcodes, because it is just a very specific helper script. Yet, I would want it to be distributed easily to all stations, and the community can also benefit from it. I think putting it into the `utils` folder is a good compromise (even though many other modules here do not qualify as utils...)
It does not really upgrade any schema, it only creates one if none has been there. I think the whole thing is bare convenience and we could thus leave it as it is, but we should extend the docstring to capture that side effect.
But as said above I don't have a strong preference for that solution.

Reply:
agree. let's leave as is.